src/analyses/mallocWrapperAnalysis.ml

```diff
@@ -13,21 +14,61 @@ struct
       let bot_name = "Unreachable node"
     end)
 
-  module Node = struct
-    include Node
-    (* Description that gets appended to the varinfo-name in user output. *)
-    let describe_varinfo (v: varinfo) node =
+  module Chain = Lattice.Chain (struct
+      let n =
+        let p = get_int "ana.malloc.unique_address_count" in
+        if p < 0 then
+          failwith "Option ana.malloc.unique_address_count has to be non-negative"
+        else p + 1 (* Unique addresses + top address *)
+
+      let names x = if Int.equal x (n - 1) then "top" else Format.asprintf "%d" x
+
+    end)
+
+  (* Map for counting malloc node visits up to n. *)
+  module MallocCounter = struct
+    include MapDomain.MapBot_LiftTop(PL)(Chain)
+
+    (* Increase counter for given node. If it does not exists yet, create it. *)
+    let add_malloc counter node =
+      let malloc = `Lifted node in
+      let count = find malloc counter in
+      if Chain.is_top count then
+        counter
+      else
+        remove malloc counter
+        |> add malloc (count + 1)
+  end
+
+  module Domain = struct
+    include Lattice.Prod (MallocCounter) (PL)
+
+    let join (counter1, node1) (counter2, node2) =
+      (MallocCounter.join counter1 counter2, node1)
+
+    let has_wrapper_node (_, wrapper_node) = not @@ PL.is_top wrapper_node
```

Comment:
`wrapper_node` is from a flat lattice of nodes, so it could also be bottom (no malloc wrapper present). This function currently returns `true` then, which doesn't sound right.